### DISCUSSION TITLE
Inconsistent Text Representation in Different App Versions

### DISCUSSION THREAD

**Discussion by hassineabd** [appium/appium#20480] [General]
I'm encountering an issue with text representation between different versions of my app when using Appium. 
Below are the details:

**Problem Description**
In some versions of my app, the text is represented as follows:

`<XCUIElementTypeStaticText type="XCUIElementTypeStaticText" value="Bienvenue à bord de l'application my-app-name !" name="Bienvenue à bord de l'application my-app-name !" label="Bienvenue à bord de l'application my-app-name !" enabled="true" visible="true" accessible="true" x="32" y="590" width="350" height="53" index="0"/>`

However, in some other versions, the same text appears differently:

`<XCUIElementTypeStaticText type="XCUIElementTypeStaticText" value="Bienvenue à bord de l'application&#10;my-app-name !" name="Bienvenue à bord de l'application&#10;my-app-name !" label="Bienvenue à bord de l'application&#10;my-app-name !" enabled="true" visible="false" accessible="true" x="879" y="522" width="312" height="50" index="7"/>`

--> "\&\#10;" is appeared instead of white space

This discrepancy does not appear in Appium Inspector, which always shows the text without any encoded line breaks or differences.

Environment Details
Appium Version: 2.5.4
iOS Version and Device: All devices and iOS version
XCUITest version: tried with multiple version and issue still appears

- The developers have confirmed that there were no changes made between the versions of the app in terms of UI framework, layout, or other relevant updates. 
- The text representation should be consistent across all versions, whether it includes those characters (like &#10;) or not.

**Screenshot** 
![screenshot idfm (1)](https://github.com/user-attachments/assets/f380f920-fe5d-424d-b819-6c1b444197d1)

**Comment by KazuCocoa** [comment]
Please attach the appium server log as well. The result usually depends on XCTest framework, but for example new lines could be parsed for XML.

**Reply by hassineabd** [reply]
Thanks for your attention 

The problem is that i am using same stack & framework , same versions 
Python, Robot framework , appium ios driver XCUITest , 

I will attach appium server logs ASAP
